vrf.go

```diff
@@ -60,13 +60,33 @@ func (out *VrfOutput) Encode() [32]byte {
 	return outbytes
 }
 
+// Decode sets the VrfOutput to the decoded input
+func (out *VrfOutput) Decode(in [32]byte) {
+	output := r255.NewElement()
+	output.Decode(in[:])
+	out.output = output
+}
+
 // Encode returns a 64-byte encoded VrfProof
-func (p *VrfProof) Encode() []byte {
+func (p *VrfProof) Encode() [64]byte {
 	cbytes := [32]byte{}
 	copy(cbytes[:], p.c.Encode([]byte{}))
 	sbytes := [32]byte{}
 	copy(sbytes[:], p.s.Encode([]byte{}))
-	return append(cbytes[:], sbytes[:]...)
+	enc := [64]byte{}
+	copy(enc[:32], cbytes[:])
+	copy(enc[32:], sbytes[:])
+	return enc
+}
+
+// Decode sets the VrfProof to the decoded input
+func (p *VrfProof) Decode(in [64]byte) {
+	c := r255.NewScalar()
+	c.Decode(in[:32])
```

Comment:
Perhaps we should return the errors from `c.Decode()`? It's also used in some of the existing code above without error checks

Reply:
probably a good move, will update